services/spd/trusty/trusty.c

```diff
@@ -363,18 +363,18 @@ static void trusty_cpu_on_finish_handler(uint64_t unused)
 	if (!ctx->saved_sp) {
 		trusty_init();
 	} else {
-		trusty_cpu_resume();
+		trusty_cpu_resume(1);
 	}
 }
 
 static void trusty_cpu_suspend_handler(uint64_t unused)
 {
-	trusty_cpu_suspend();
+	trusty_cpu_suspend(0);
 }
 
 static void trusty_cpu_suspend_finish_handler(uint64_t unused)
 {
-	trusty_cpu_resume();
+	trusty_cpu_resume(0);
 }
 
```

Comment:
In general does it make sense to pass the actual suspend state to Trusty? I suppose TAs can definitely use the System Suspend state. Im not sure how that case would be handled with just two status values

Reply:
Do we have access to the system suspend state here? There is a separate psci call for system suspend, but that information seems to be lost before these hooks are called.

Reply:
I see that system_suspend eventually calls into this trusty_cpu_suspend() with the max suspend level. So trusty SPD should be able to differentiate easily, AFAICS.

Reply:
We can add more values later if we need to know about system suspend. For now I don't want to add a hack to try to guess if we are entering system suspend based on the platform specific power level.